src/components/tasks/card/AssigneeDropdownMenu.tsx

```diff
@@ -0,0 +1,96 @@
+import { FC, useState, useEffect, SetStateAction } from "react";
+import userType from '@/interfaces/user.type';
+import { ALL_USERS } from '@/components/constants/url';
+import fetch from '@/helperFunctions/fetch';
+import { toast, ToastTypes } from '@/helperFunctions/toast';
+import classNames from '@/components/tasks/card/card.module.scss';
+
+type Props = {
+  cardDetails: any
+  updateAssignee: (username: string) => void
+  setisAssigneeDropdownOpen: React.Dispatch<SetStateAction<boolean>>
+}
+
+//https://github.com/Real-Dev-Squad/website-status/issues/303
+
+const AssigneeDropdownMenu: FC<Props> = ({ cardDetails,updateAssignee,setisAssigneeDropdownOpen }) => {
+  const [searchValue, setSearchValue] = useState("");
+  const [filteredList, setFilteredList] = useState([]);
+  const [assigneeList, setAssigneeList] = useState([])
+  const { ERROR } = ToastTypes;
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const { requestPromise } = fetch({ url: ALL_USERS });
+        const { data } = await requestPromise;
+        const usersData = data.members || data.users
+        setAssigneeList(usersData)
+      } catch (err: any) {
+        toast(ERROR, err.message);
+      }
+    };
+    fetchData();
+
+  }, []);
+
+  useEffect(() => {
+    if (searchValue) {
+      const newFilteredList = assigneeList.filter((member: userType) => (
+        member.username?.toLowerCase().includes(searchValue.toLowerCase())
+      ));
+      setFilteredList(newFilteredList)
+    }
+    else {
+      setFilteredList([])
+    }
+  }, [searchValue])
+
+  const onChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchValue(e.target.value)
```

Comment:
add a debounce here so that we can avoid unwanted network calls and re-renders

Reply:
Great point @ivinayakg , however i didnt see any network calls happening over here

Reply:
I'm only making one network request on component mounts, we are storing the data in the application memory unless the page refreshes it won't make a network call and yes @ivinayakg it's not a network call and @harshith-venkatesh not sure if you remember i have asked you about this on discord in a discussion, you have mentioned we don't have dynamic data of users which keeps changing frequently and it's a JSON(lightweight) with very less we can make one request call and save state as ```assigneeList``` istead of making a debounce call for search value change

Reply:
My bad, yes there aren't any network calls but still add a debounce as there are still going to be unnecessary re-renders